app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -372,17 +372,19 @@ class WallpapersUseCases(
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal class DefaultLoadBitmapUseCase(private val context: Context) : LoadBitmapUseCase {
+    internal class DefaultLoadBitmapUseCase(
+        private val filesDir: File,
+        private val orientation: Int,
```

Comment:
These are lazy singletons. So doesn't that mean the first orientation that is queried will be used for that application lifecycle?

Reply:
@jonalmeida I was under the impression that this object will be recreated on configuration changes. I checked and it is indeed singleton. Does this mean, we would not be able to decouple context and this class?

Reply:
I think you can decouple it from the class constructor but will probably need to keep it in the method parameters to query at the time of use for the orientation.

Reply:
If we pass it in the method, then we will need to change LegacyBitmapUseCase.invoke() as well (apply wallpaper method of HomeFragment assumes invoke() of both Legacy and Default UseCase to follow the same signature  ). Should I go ahead and make the change?

Reply:
Rather than update the `invoke` function, I would prefer if we changed the `orientation` parameter sent to `DefaultLoadBitmapUseCase` to be a functional parameter. That way, the `Context` can be re-queried for the current orientation every time `invoke` is run. So something like:

```
    DefaultLoadBitmapUseCase(context.filesDir, { context.resources.configuration.orientation })

    internal class DefaultLoadBitmapUseCase(
        private val filesDir: File,
        private val getOrientation: () -> Int,
    )
```

Reply:
@MatthewTighe I have updated the PR with your suggestion.